components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -164,8 +178,12 @@ public ResendConfirmationDTO resendConfirmation(String tenantDomain, String rese
         } else {
             userRecoveryDataStore.invalidate(user);
             confirmationCode = Utils.generateSecretKey(notificationChannel, user.getTenantDomain(), recoveryScenario);
+            if (recoveryFlowId != null && !StringUtils.equals(notificationChannel,
+                    NotificationChannels.SMS_CHANNEL.getChannelType())) {
+                confirmationCode = recoveryFlowId + "." + confirmationCode;
```

Comment:
define a constant for the separator and use it when combining the values and decoupling the values

Reply:
Fixed with bc26f3b539ca0f4b6b3c8a3b52fc7fcd619872ac